Release 1.3 (M4 Fall25)

FabrizioMoggio opened the pull request:
#### What type of PR is this?
* documentation

#### What this PR does / why we need it:
This is the M4 PR for the public version of the EFN API, for the Fall25 Meta Release. The PR changes the API version from WIP to 0.1.0 ad provides the updated README.md and CHANGELOG.md files.

#### Which issue(s) this PR fixes:
https://github.com/camaraproject/EnergyFootprintNotification/issues/100

hdamker commented:
@FabrizioMoggio I've done first round of review and provide here the results. Please do all needed changes which are not directly related to the release within separate PRs and update the release PR branch afterwards from main.

### API Name - critical:

* I understand that you changed the API name during the review of rc.1 by @tanjadegroot from energy-footprint-notification to energy-footprint
* But the change hasn't been done only partly:
  * file names of the API and test definitions should use the API name, they are still energy-footprint-notification
  * the CHANGELOG.md still lists energy-footpring-notification
  * the CloudEventType uses still energy-footpring-notification as api-name
  * the test definition is still using the previous server URL (and is therefore inconsistent
* My recommendation would be to revert back to `energy-footprint-notification` as API name. Why:
  * Asynchronous, callback-based API; “notification” clarifies the delivery model
  * Avoids ambiguity of “energy-footprint” (which suggests synchronous query)
  * Already consistent with repository name, file names, CloudEvent types and test definitions -> less alignments needed across all files
* Alignments to make:
  * Servers base path: {apiRoot}/energy-footprint-notification/v0.1
  * OAuth scopes: energy-footprint-notification:calculate-energy-consumption, energy-footprint-notification:calculate-carbon-footprint

### Callbacks/Events

* They "MUST use the media type application/cloudevents+json" (currently "application/json"), see https://github.com/camaraproject/Commonalities/blob/main/documentation/CAMARA-API-Event-Subscription-and-Notification-Guide.md#31-event-notification-definition

### Info.description

- In info.description, remove outdated endpoint bullets (e.g., /overall-energy-consumption) that do not match actual paths (/calculate-energy-consumption, /calculate-carbon-footprint).

hdamker commented on CHANGELOG.md:
```diff
@@ -15,6 +16,43 @@ The below sections record the changes for each API version in each release as fo
+### Removed
+
+
+**Full Changelog**: https://github.com/camaraproject/EnergyFootprintNotification/compare/r1.1...r1.3
```
The section for a public release collects all changes since the last public release, so in this case since the start of the repository:

```suggestion
**Full Changelog**: https://github.com/camaraproject/EnergyFootprintNotification/compare/commits/r1.3
```

FabrizioMoggio replied:
implemented

hdamker commented on CHANGELOG.md:
```diff
@@ -15,6 +16,43 @@ The below sections record the changes for each API version in each release as fo
+  - [View it on ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/EnergyFootprintNotification/r1.3/code/API_definitions/energy-footprint-notification.yaml&nocors)
+  - [View it on Swagger Editor](https://camaraproject.github.io/swagger-ui/?url=https://raw.githubusercontent.com/camaraproject/EnergyFootprintNotification/r1.3/code/API_definitions/energy-footprint-notification.yaml)
+
+### Added
+* Support for the following use cases: https://github.com/camaraproject/EnergyFootprintNotification/discussions/11
+  
+### Changed
+* Renamed operations: https://github.com/camaraproject/EnergyFootprintNotification/pull/83 
+* Changed OAuth2 with OpenId: https://github.com/camaraproject/EnergyFootprintNotification/pull/83
+* Updated the name of the test file to be coherent with the operationId: https://github.com/camaraproject/EnergyFootprintNotification/pull/83
+* info.description reordering: https://github.com/camaraproject/EnergyFootprintNotification/pull/92
+* ErrorInfo update according to Commonalities 6.0 public release: https://github.com/camaraproject/EnergyFootprintNotification/pull/95
+
+### Fixed
+* operationId calculatEnergyConsumption: https://github.com/camaraproject/EnergyFootprintNotification/issues/88
```
As this is the changelog for the public release actually everything is new, but also fine to list the latest changes. But would make sense to add a line to which pre-release you are comparing.

hdamker commented:
Note: the request for changes is mainly related to the issues in https://github.com/camaraproject/EnergyFootprintNotification/pull/98#issuecomment-3239415698.
Please update the PR from main when the needed alignments are done.

FabrizioMoggio commented:
> * CloudEventType

@hdamker I implemented the removal of the "Notification" before reading your suggestion of moving back to the old name. I'm open to any suggestion If you still think EFN is more coherent than EF, I'

> @FabrizioMoggio great to see that you working on the issues, but PLEASE don't do it here in the release PR. Make a clean new PR against main, and update afterwards the release PR from `main`.
> 
> BTW another issue you might want to address, but minor criticality:
> 
> * Inconsistent casing for request identifier field across spec/examples/tests:
>   
>   * OAS response schema uses `requestId` (`components.schemas.ReportCreationRequest`), CloudEventEnergy example uses `requestId`, but CloudEventCarbonFootprint schema uses `requestID`.
>   * Feature files also assert `$.requestID`.
>   * Fix: standardize on one key (recommend `requestId`) across OAS schemas, examples, and feature assertions.

Dear @hdamker I had already implemented your suggestion in the Release PR before noticing this message. Is there any suggestion on how to move forward now?

hdamker commented:
> Dear @hdamker I had already implemented your suggestions in the Release PR before noticing this message. Is there any suggestion on how to move forward now? I could (and this should be aligned with your suggestion in my understanding):
> 
> * create a new wip PR
> * merge in main
> * in my branch that is the origin for the M4 release PR, align with main so to have the M4 release PR with just the formal requested modifications, e.g. the release number

I won't insist on it, but a way to "clean" the release PR could be indeed:
* create a new PR in main for the review comments which are not related to the release
* cherry pick in this PR the commits which you have done here
* merge this PR into main after review
* merge main into the PR here ... with that the changes would "go away", without the need to remove or revert them first

hdamker commented:
If you plan to run the PR with non-release changes first, you might want to consider to set the PR here to draft until done.

hdamker commented:
> I have already done the WIP PR (#103), now this PR should be ok and ready to be reviewed/approved

@FabrizioMoggio yes, much nicer now. Even our release validation [workflow is happy](https://github.com/camaraproject/ReleaseManagement/actions/runs/17410993982). Will continue with the review.

hdamker commented:
Beyond the above comments there is a fix of the ErrorInfo schema in CAMARA_common.yaml which will come in a patch release of Commonalities. If you apply it already now you will be compliant to r3.3 of the API Design Guide, but it is neither critical nor mandatory for you now: 

- https://github.com/camaraproject/Commonalities/pull/517 (issue #515)

If you apply it now then please again in a separate PR against main.

hdamker commented:
> If not mandatory, at this point, I prefer not to engage the supporting partners on another round of wip-approval-merge-align- etc. This would be easy for this API but for TI API and CFS API I need more coordination with the other members. For this reason I'm planning to keep this issue open for all my APIs and fix it for the next meta release. The time is a great constraint at this point.

@FabrizioMoggio That's fine and your decision. But just seen that in TrafficInfluence you need anyway to clean-up your legacy ErrResponse schema(s).

FabrizioMoggio commented:
> > If not mandatory, at this point, I prefer not to engage the supporting partners on another round of wip-approval-merge-align- etc. This would be easy for this API but for TI API and CFS API I need more coordination with the other members. For this reason I'm planning to keep this issue open for all my APIs and fix it for the next meta release. The time is a great constraint at this point.
> 
> @FabrizioMoggio That's fine and your decision. But just seen that in TrafficInfluence you need anyway to clean-up your legacy ErrResponse schema(s).

If I have other issues to fix I will also include it.

FabrizioMoggio commented:
ErroInfo fixed

hdamker commented:
> ErroInfo fixed

Unfortunately not ... you changed only the order in "required", but not the actual schema.

hdamker commented:
> Everything has been addressed

Great! I will approve now for release management. You might want to consider to remove the placeholder README.md files in the directories. Happy to approve again, if you dismiss my approval with that.

hdamker reviewed (approved):
Approved on behalf of Release Management :clap:

Next steps for the team:

- [x] PR merged (by API repository codeowner)
- [x] Release created within GitHub (by API repository codeowner)
- [ ] Release Tracker updated (with creation date of the release and the release tag link)

hdamker commented:
> Dear Herbert, I'm sorry I didn't notice this request of yours before creating the release.

No worries, the files are doing no harm. Let's delete them together with the reset to `wip`.

FabrizioMoggio commented:
> > Dear Herbert, I'm sorry I didn't notice this request of yours before creating the release.
> 
> No worries, the files are doing no harm. Let's delete them together with the reset to `wip`.

I'm sorry I'm not sure how to proceed because I'm not sure to have got you right :-). The release has been merged and the release tag created. I will delete the file when a new wip will be created. For the next release I mean. Is my understanding right?
